postcommit.py

```diff
@@ -0,0 +1,642 @@
+    	def parseCommits(self, commitlog, branch):
+        	commit_pattern = re.compile('CMHash:([a-z0-9]+) ([a-z0-9]+)##CMParent:([a-z0-9\s]+)?##([a-z0-9\s]+)?##CMAuthor:([a-zA-Z0-9.\-@+.\s]+)##CMDate:([0-9:\s+-]*)##CMMessage:\n*((?:.|\n)*?)\n*LOGENDS\n+(([0-9]+\s+(.*?)\n)*)?', re.DOTALL)
+        	commits = commit_pattern.findall(commitlog)     	
+
+        	for commit in commits:
+           		node = commit[0]
+            		revision = commit[1]
+            		parents = commit[2].split(' ')
+            		parents_short = commit[3].split(' ')
+            		username = commit[4]
+            		commit_date = re.sub('\+0530', '', commit[5]).strip()
+            		description = striplicense_template(commit[6])
+            		commited_files = commit[7].strip('\n').splitlines()
+            		commit_type = 1
+            		parentChangeset1 = parents[0]
+            		parent1_branch = parents[0]
+            		self.newchangesets.append(node)
+
+            		log = {'changeset': node[:12], 'revision': revision, 'branch': branch, 'user': username,
+                  	       'message': description, 'commit_type': commit_type, 'parents': [],
+                   	       'commit_date': commit_date, 'push_time': getCurrentTime()
+                   	      }  
+
+			#  diff_changes temporarly not going to be used *may* be useful in future (just add **self.getDiffContent(parent,node)** in diff_changes )
+            		for parent in parents:
+                		log['parents'].append({'changeset' :parent[:12], 'branch' : parent[:12],
+                                      		       'diff_changes' : ""})
+            		if len(parents) == 2:
+                		log = self.updateMergeDetails(log, parents, parents_short)
+                		log['commit_type'] = 2
+
+            		log['files'] = self.filestatus(node)
+	    
+	    		self.logs.append(log)
+
+#	***************************************************************************************************************
+#	getDiffContent :
+#
+#			Show changes between two changeset
+#
+#			@param : changeset1,changeset2
+#			
+#			@return : [{'changed_lines': [1,4,6], 'file': 'test'}]
+#       *************************************************************************************************************** 
+
+    	def getDiffContent(self, changeset1, changeset2):
+        	diff_command = "git diff -U0 " + changeset1 + " " + changeset2 
+        	result = status_check(diff_command, self.module_dir)
+        	diff_changes = self.parseDiffContent(list(result))
+        	return diff_changes
+
+
+#	***************************************************************************************************************
+#	getDiffLines :
+#
+#			Find number of lines newly added in file
+#
+#			@param : changes
+#
+#			@return : [1,2,4,5]
+#       ***************************************************************************************************************
+    	def getDiffLines(self,changes):
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename method "getDiffLines" to match the regular expression ^[a-z_][a-z0-9_]{2,}$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS100)